src/scripts/h5p-dictation-sentence.js

```diff
@@ -406,16 +412,42 @@ class Sentence {
     // Add spaces to correct text
     const wordsSolution = Sentence.addSpaces(
       this.getCorrectText(),
-      this.params.autosplit
-    ).split(' ');
+      this.params.autosplit,
+      this.params.wordSeparator
+    ).split(this.params.wordSeparator);
 
     let input = this.getUserInput();
     if (this.params.ignorePunctuation) {
       input = Sentence.stripPunctuation(input);
     }
 
-    // Add spaces to solution
-    const wordsInput = input.trim() === '' ? [] : Sentence.addSpaces(input, this.params.autosplit).split(' ');
+    // In case our wordSeparator is not space, we must replace spaces by the separator
+    if(' ' !== this.params.wordSeparator) {
+      // If we have one alternative with multiple words, we want to escape the spaces
+      wordsSolution.forEach( solutionPart => {
+        const alternatives = solutionPart.split('|');
+        
+        alternatives.forEach(alternative => {
+          alternative = alternative.trim();
+          if(alternative.includes(' ')) {
+            const escapedAlternative = alternative.replaceAll(' ', Sentence.SPACE_ESCAPE);
+            input = input.replaceAll(alternative, escapedAlternative)
+          }
+        });
+      });
+      
+      // And then unescape spaces
+      input = input.replaceAll(" ", this.params.wordSeparator);
+      input = input.replaceAll(Sentence.SPACE_ESCAPE, ' ');
+    }
+
+    // Add spaces to solution and break in parts
+    let wordsInput = input.trim() === '' ? [] : Sentence.addSpaces(input, this.params.autosplit, this.params.wordSeparator).split(this.params.wordSeparator).filter(word => word.length>0);
+    
+    // In case our wordSeparator is not space, we add spaces between tokens
+    if(' ' !== this.params.wordSeparator) {
+      wordsInput = wordsInput.map( (word,i) => i==wordsInput.length-1 ? word : word + ' ');
```

Comment:
The H5P core team favors template-strings nowadays as Babel takes care of those for IE11 even without a polyfill.